posts/2019-10-30-Async-await-stable.md

```diff
@@ -0,0 +1,146 @@
+---
+layout: post
+title: "Async-await on stable Rust!"
+author: Niko Matsakis
+---
+
+**On this coming Thursday, November 7, async-await syntax hits stable
+Rust, as part of the 1.39.0 release.** This work has been a long time
+in development -- the key ideas for zero-cost futures, for example,
+were [first proposed by Aaron Turon and Alex Crichton in
+2016][zcf-rust]! -- and we are very proud of the end result. We believe
+that Async I/O is going to be an increasingly important part of Rust's
+story.
+
+While this first release of "async-await" is a momentous event, it's
+also only the beginning. The current support for async-await marks a
+kind of "Minimum Viable Product" (MVP). We expect to be polishing,
+improving, and extending it for some time.
+
+Already, in the time since [async-await hit beta][aa-beta], we've made
+a lot of great progress, including making some [key diagnostic
+improvements][diag] that help to make async-await errors far more
+approachable. To get involved in that work, check out
+the [Async Foundations Working Group][wg]; if nothing else, you can
+help us by filing bugs about polish issues or by [nominating those
+bugs that are bothering you the most][nom], to help direct our
+efforts.
+
+# Major developments in the async ecosystem
+
+Now that async-await is approaching stabilization, all the major Async
+I/O runtimes are at work adding and extending their support for the
+new syntax:
+
+* the [tokio] runtime [recently announced a number of scheduler
+  improvements][tokio-sched], and they are planning a stable release
+  in November that supports async-await syntax;
+* the [async-std] runtime [has been putting out weekly releases for the past few months][as-releases], and plans to make their
+  1.0 release shortly after async-await hits stable;
+* using [wasm-bindgen-futures], you can even bridge Rust Futures with
+  [JavaScript promises];
+* the [hyper library][hyper] has [migrated][hyper#1805] to adopt standard Rust futures;
+* the 0.3.0 version of the [futures-rs library][futures] will support
+  async-await and will be released at or around the time async-await hits stable
+  (you can use the [0.3.0-alpha][] releases now);
+* finally, async-await support is starting to become available in higher-level
+  [web frameworks][wf] as well.
+
```

Comment:
Feel free to mention [futures-intrusive](https://docs.rs/futures-intrusive/0.2.0/futures_intrusive/) as a novel user of async/await and pinning that allows for 100% allocation free async synchronization primitives (channels, mutexes, etc). I don't really like self-promotion, but OTOH I think the concept is cool that it might be worth mentioning it.  It was also only possible to implement due to the properties of async/await in Rust.